docs/usage/types.md

```diff
@@ -660,3 +660,30 @@ to get validators to parse and validate the input data.
 {!.tmp_examples/custom_data_types.py!}
 ```
 _(This script is complete, it should run "as is")_
+
+## Parsing data into a specified type
```

Comment:
I think this should be in `models.md` or a new usage file.

Reply:
Yeah, I think it definitely makes sense to create a new usage file if/when we add `dump_as_type`, but it was hard for me to justify a new docs file for just this short little section.

I'll move it to `models.md` for now.